src/riak_core_bucket.erl

```diff
@@ -192,14 +197,61 @@ bucket_nval_map(Ring) ->
 default_object_nval() ->
     riak_core_bucket:n_val(riak_core_bucket_props:defaults()).
 
+%% @private
+%% @doc For testing/debugging
+-spec all_n() -> [pos_integer(),...].
+all_n() ->
+    {ok, Ring} = riak_core_ring_manager:get_my_ring(),
+    all_n(Ring).
+
+-spec all_n(riak_core_ring()) -> [pos_integer(),...].
 all_n(Ring) ->
-    BucketNs = bucket_nval_map(Ring),
-    DefaultN = default_object_nval(),
-    AllN = lists:foldl(fun({_, N}, Acc) ->
-                               ordsets:add_element(N, Acc)
-                       end, [DefaultN], BucketNs),
+    BucketNVals = bucket_nvals(Ring),
+    BucketTypeNVals = bucket_type_nvals(),
+    ordsets:from_list(
```

Comment:
I think the `ordsets:from_list` call is redundant: that function takes an unordered list and returns an ordset (which is implemented as an ordered list). But `ordsets:union` should already be returning an ordset.